src/main/scala/com/softwaremill/mqperf/mq/ArtemisMq.scala

```diff
@@ -0,0 +1,18 @@
+package com.softwaremill.mqperf.mq
+
+import javax.jms.ConnectionFactory
+
+import com.softwaremill.mqperf.config.TestConfig
+import org.apache.activemq.artemis.jms.client.ActiveMQConnectionFactory
+
+class ArtemisMq(val testConfig: TestConfig) extends Jms2Mq {
+
+  override lazy val connectionFactory: ConnectionFactory = {
+    val hosts = "failover:(" + testConfig.brokerHosts.map(h => s"tcp://$h:61616").mkString(",") + ")"
+    val cf = new ActiveMQConnectionFactory(hosts)
+    cf.setConfirmationWindowSize(10485760)
+    cf.setBlockOnDurableSend(false)
```

Comment:
>send+CompletionListener+blockOnDurableSend=true behave?

if transactional it will not block still as thats managed by another setting which is default false.
if non-transactional it will block on send, and like wise it will also invoke completion listener on completion.

Thats fair enough. Still clustered setup is in the docs and actually is more similar to a kafka setup, atm you're doing single master.  (the advantage on this is where as kafka has one deployment model, artemis (and rabbit mq and activemq for that fact) support several so if specific needs is better fit.

I should mention we use in my org both kafka and artemis, but we use them complimentary to each other (we actually bridge the two), i def think kafka is awesome, just want to help you get more accurate results for artemis, with using JMS 2.0.